plugins/woocommerce-admin/client/products/sections/pricing-section.tsx

```diff
@@ -135,10 +132,8 @@ export const PricingSection: React.FC = () => {
 							{ ...regularPriceProps }
 							label={ __( 'List price', 'woocommerce' ) }
 							placeholder={ __( '10.59', 'woocommerce' ) }
-							onChange={ ( value: string ) => {
-								const sanitizedValue = sanitizePrice( value );
-								regularPriceProps?.onChange( sanitizedValue );
-							} }
+							value={ formatCurrencyDisplayValue( regularPriceProps?.value, currencyConfig, formatAmount ) }
+							onBlur={ () => setValue( 'regular_price', sanitizePrice( regularPriceProps.value ) ) }
```

Comment:
@octaedro I moved the sanitization to `onBlur`.  I think this is a little bit more conventional as I was worried about input being sanitized as it was typed could lead to odd behavior.

But please let me know if I broke anything here or if this needs to be sanitized as we type.

Reply:
Sounds good to me! Good idea!